src/cluster_legacy.c

```diff
@@ -2554,45 +2622,49 @@ static clusterMsgPingExt *getNextPingExt(clusterMsgPingExt *ext) {
 }
 
 /* All PING extensions must be 8-byte aligned */
-uint32_t getAlignedPingExtSize(uint32_t dataSize) {
+static uint32_t getAlignedPingExtSize(uint32_t dataSize) {
     return sizeof(clusterMsgPingExt) + EIGHT_BYTE_ALIGN(dataSize);
 }
 
-uint32_t getHostnamePingExtSize(void) {
-    if (sdslen(myself->hostname) == 0) {
-        return 0;
-    }
-    return getAlignedPingExtSize(sdslen(myself->hostname) + 1);
-}
-
-uint32_t getHumanNodenamePingExtSize(void) {
-    if (sdslen(myself->human_nodename) == 0) {
-        return 0;
-    }
-    return getAlignedPingExtSize(sdslen(myself->human_nodename) + 1);
-}
-
-uint32_t getShardIdPingExtSize(void) {
+static uint32_t getShardIdPingExtSize(void) {
     return getAlignedPingExtSize(sizeof(clusterMsgPingExtShardId));
 }
 
-uint32_t getForgottenNodeExtSize(void) {
+static uint32_t getForgottenNodeExtSize(void) {
     return getAlignedPingExtSize(sizeof(clusterMsgPingExtForgottenNode));
 }
 
-void *preparePingExt(clusterMsgPingExt *ext, uint16_t type, uint32_t length) {
+static void *preparePingExt(clusterMsgPingExt *ext, uint16_t type, uint32_t length) {
     ext->type = htons(type);
     ext->length = htonl(length);
     return &ext->ext[0];
 }
 
+/* If value is nonempty and cursor_ptr points to a non-NULL cursor, writes a
+ * ping extension at the cursor, advances the cursor, increments totlen and
+ * returns 1. If value is nonempty and cursor_ptr points to NULL, just computes
+ * the size, increments totlen and returns 1. If value is empty, returns 0. */
+static uint32_t
+writeSdsPingExtIfNonempty(uint32_t *totlen_ptr, clusterMsgPingExt **cursor_ptr, clusterMsgPingtypes type, sds value) {
+    size_t len = sdslen(value);
+    if (len == 0) return 0;
+    size_t size = getAlignedPingExtSize(len + 1);
+    if (*cursor_ptr != NULL) {
+        void *ext = preparePingExt(*cursor_ptr, type, size);
+        memcpy(ext, value, len);
+        *cursor_ptr = getNextPingExt(*cursor_ptr);
+    }
+    *totlen_ptr += size;
+    return 1;
+}
+
 /* 1. If a NULL hdr is provided, compute the extension size;
- * 2. If a non-NULL hdr is provided, write the hostname ping
- *    extension at the start of the cursor. This function
+ * 2. If a non-NULL hdr is provided, write the ping
+ *    extensions at the start of the cursor. This function
  *    will update the cursor to point to the end of the
  *    written extension and will return the amount of bytes
  *    written. */
-uint32_t writePingExt(clusterMsg *hdr, int gossipcount) {
+static uint32_t writePingExtensions(clusterMsg *hdr, int gossipcount) {
```

Comment:
unrelated to this PR but more of a global coding style question - what do you think we start using the `_` prefix for all internal data structures and functions? If @valkey-io/core-team is aligned, we can do it in a separate PR (or we can do it incrementally too).

```suggestion
static uint32_t _writePingExtensions(clusterMsg *hdr, int gossipcount) {
```

Reply:
No, I don't like that underscore prefix style actually. I was going to comment on that in your PR. :) We have `static` for this purpose. The public/private property doesn't need to be in the name.

Reply:
I didn't start the `_` prefix :-) but I kind of like it since I don't need to go look for `static` in the prototype, which might be at a different location. But I can live without `_` too. Consistency is more important.

Reply:
Yeah, we have several different styles in different areas and files. Let's talk about it in a meeting.

When or why do you need to immediately see if a function is private or public?

For some time, Salvatore didn't use static because static functions didn't show up in stack traces (before `-fno-omit-stack-frame` existed, or in crash reports from users?), so maybe he used `_` instead of static. In Python, there's no way to mark anything as private so they use `_` by convention. But we do use static now.

I think this underscore style is a special case of [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), where various things, like type, is encoded in variable and function names. I'm not a fan of that. If we want to change the type or the visibility of a function, we don't need to change the name and update all the calls sites. We do use prefixes like `is` and `has` for boolean-ish functions, plural for functions returning lists, etc. but this is in a readable way.